src/compiler/core.ts

```diff
@@ -2630,4 +2630,8 @@ namespace ts {
     export function and<T>(f: (arg: T) => boolean, g: (arg: T) => boolean) {
         return (arg: T) => f(arg) && g(arg);
     }
+
+    export function never(_: never): never {

```

Comment:
Is this helper really necessary? If so, consider converting the "unreachable" throws in `checker.ts` to call this.